### PR TITLE
feat: studio mobile nav

### PR DESCRIPTION
**Pull request by fsansalvadore**
https://github.com/user-attachments/assets/a2e1c815-6d65-486f-ae77-1d5993377c40

### CODE REVIEW

File: apps/studio/pages/project/[ref]/index.tsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ import { useAppStateSnapshot } from 'state/app-state'
 import type { NextPageWithLayout } from 'types'
 import {
   Badge,
+  Button,
```
Comment (joshenlim) [inline apps/studio/pages/project/[ref]/index.tsx:22]: nit - seems like unnecessary import

File: packages/common/hooks/useDragToClose.tsx [hunk 2]
```diff
@@ -16,6 +16,7 @@ const useDragToClose = ({
 
   const handleTouchStart = useCallback(
     (e: React.TouchEvent<HTMLDivElement>) => {
+      console.log('Touch started')
```
Comment (joshenlim) [inline packages/common/hooks/useDragToClose.tsx:19]: remove console log

File: apps/studio/pages/project/[ref]/index.tsx (hunk 1 above)
Comment (fsansalvadore) [reply apps/studio/pages/project/[ref]/index.tsx:22]: oh yeah, thanks!

**Comment by joshenlim** [comment]
feel free to address this in another PR: incident banner a tad squished on mobile
image

**Review by joshenlim — APPROVED** [review]
LGTM - did a smoke test in general for both desktop and mobile
couple of issues on mobile, but will be addressed in subsequent PRs

**Comment by raw1359** [comment]
H

**Review by raw1359 — APPROVED** [review]

**Comment by fsansalvadore** [comment]
@raw1359 thank you sir

File: apps/studio/components/layouts/ProjectLayout/ProjectLayout.tsx [hunk 3]
```diff
@@ -168,18 +195,23 @@ const ProjectLayout = forwardRef<HTMLDivElement, PropsWithChildren<ProjectLayout
...
-                className={cn({ hidden: !showProductMenu || !productMenu })}
+                className={cn('hidden md:flex', { '!hidden': !showProductMenu || !productMenu })}
                 withHandle
                 disabled={resizableSidebar ? false : true}
               />
               <ResizablePanel id="panel-right" className="h-full flex flex-col">
-                {!hideHeader && IS_PLATFORM && <LayoutHeader />}
+                {!hideHeader && IS_PLATFORM && (
+                  <LayoutHeader
+                    showProductMenu={!!(showProductMenu && productMenu)}
```
Comment (MildTomato) [inline apps/studio/components/layouts/ProjectLayout/ProjectLayout.tsx:205]: can we stop using props for things like this?
should we not be using app state?
@joshenlim @fsansalvadore 

I'm in another branch and I'm probably going to remove these props if that's ok.

Comment (MildTomato) [reply apps/studio/components/layouts/ProjectLayout/ProjectLayout.tsx:205]: also title, could have logic for the title handled inside component

**Pull request merged**